### DISCUSSION TITLE
Support Cloudflare request.cf to get to timezone and other geo data

### DISCUSSION THREAD

**Discussion by JorritSalverda** [remix-run/remix#6261] [Proposals]
With the latest package `@remix-run/cloudflare 1.15.0` in Typescript it's unfortunately not possible to use the `request.cf` property as documented at https://developers.cloudflare.com/workers/runtime-apis/request?ref=blog.cloudflare.com#incomingrequestcfproperties. It would be very helpful to be able to use it to have correct server-side rendering in the user's timezone along the lines of the following snippet:

```typescript
export const loader = async ({ request }: LoaderArgs) => {
   let client_now = DateTime.utc().setZone(request.cf.timezone);
}
```

Is it planned to add this or are there other ways to get to this property? Thx

**Comment by phuoccss** [comment]
You can access this via `context.request.cf`

**Comment by smcstewart** [comment]
Coming to this a bit late; in your `server.(j|t)s` you can change the `getLoadContext()` function to:

`getLoadContext: (context) => ({ env: context.env, cf: context.request?.cf })`

... to get access to the [`IncomingRequestCfProperties`](https://developers.cloudflare.com/workers/runtime-apis/request/#incomingrequestcfproperties) within your loaders and actions.
